src/main/java/net/neoforged/neoforge/fluids/crafting/CompoundFluidIngredient.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.fluids.crafting;
+
+import com.mojang.serialization.MapCodec;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+import net.neoforged.neoforge.common.NeoForgeMod;
+import net.neoforged.neoforge.common.crafting.CompoundIngredient;
+import net.neoforged.neoforge.fluids.FluidStack;
+
+/**
+ * Fluid ingredient that matches if any of the child ingredients match.
+ * This type additionally represents the array notation used in
+ * {@linkplain FluidIngredient#CODEC} internally.
+ *
+ * @see CompoundIngredient CompoundIngredient, its item equivalent
+ */
+public final class CompoundFluidIngredient extends FluidIngredient {
+    public static final MapCodec<CompoundFluidIngredient> CODEC = null;
```

Comment:
Hmmm? We need this for the ingredient MapCodec.

Reply:
Oh my god, *how did noone catch this* (myself included); left this as a todo for myself, then forgot to implement it :P